src/System.Web.HttpUtility/src/System/Web/HttpUtility.cs

```diff
@@ -0,0 +1,301 @@
+         )]
+        public static byte[] UrlEncodeUnicodeToBytes(string str)
+        {
+            if (str == null)
+                return null;
+            return Encoding.ASCII.GetBytes(UrlEncodeUnicode(str));
+        }
+
+        public static string UrlDecode(string str)
+        {
+            if (str == null)
+                return null;
+            return UrlDecode(str, Encoding.UTF8);
+        }
+
+        public static string UrlDecode(byte[] bytes, Encoding e)
+        {
+            if (bytes == null)
+                return null;
+            return UrlDecode(bytes, 0, bytes.Length, e);
+        }
+
+        public static byte[] UrlDecodeToBytes(string str)
+        {
+            if (str == null)
+                return null;
+            return UrlDecodeToBytes(str, Encoding.UTF8);
+        }
+
+        public static byte[] UrlDecodeToBytes(string str, Encoding e)
+        {
+            if (str == null)
+                return null;
+            return UrlDecodeToBytes(e.GetBytes(str));
+        }
+
+        public static byte[] UrlDecodeToBytes(byte[] bytes)
+        {
+            if (bytes == null)
+                return null;
+            return UrlDecodeToBytes(bytes, 0, bytes != null ? bytes.Length : 0);
+        }
+
+        public static byte[] UrlEncodeToBytes(string str, Encoding e)
+        {
+            if (str == null)
+                return null;
+            var bytes = e.GetBytes(str);
+            return HttpEncoder.UrlEncode(bytes, 0, bytes.Length, false /* alwaysCreateNewReturnValue */);
+        }
+
+        public static byte[] UrlEncodeToBytes(byte[] bytes, int offset, int count)
+        {
+            return HttpEncoder.UrlEncode(bytes, offset, count, true /* alwaysCreateNewReturnValue */);
+        }
+
+        [Obsolete(
+             "This method produces non-standards-compliant output and has interoperability issues. The preferred alternative is UrlEncode(String)."
+         )]
+        public static string UrlEncodeUnicode(string str)
+        {
+            return HttpEncoder.UrlEncodeUnicode(str, false /* ignoreAscii */);
+        }
+
+        public static string UrlDecode(string str, Encoding e)
+        {
+            return HttpEncoder.UrlDecode(str, e);
+        }
+
+        public static string UrlDecode(byte[] bytes, int offset, int count, Encoding e)
+        {
+            return HttpEncoder.UrlDecode(bytes, offset, count, e);
+        }
+
+        public static byte[] UrlDecodeToBytes(byte[] bytes, int offset, int count)
+        {
+            return HttpEncoder.UrlDecode(bytes, offset, count);
+        }
+
+        public static string JavaScriptStringEncode(string value)
+        {
+            return JavaScriptStringEncode(value, false);
+        }
+
+        public static string JavaScriptStringEncode(string value, bool addDoubleQuotes)
```

Comment:
We have a JavaScript encoder in System.Text.Encodings.Web:
https://github.com/dotnet/corefx/blob/master/src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/JavaScriptEncoder.cs
Can we just call to that and avoid having a duplicate versoin here?